src/main/java/co/cask/PluginJSON.java

```diff
@@ -256,6 +333,14 @@ private void initialize() {
       widgetDirectory = new File(widgetsDirectory);
     }
 
+    if (iconsDirectory != null) {
+      if (!iconsDirectory.contains("/")) {
+        iconDirectory = new File(project.getBasedir() + "/" + iconsDirectory);
```

Comment:
also, you should check that each directory exists and is a directory, throwing an exception if not